extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/util/ArcticCmdUtils.scala

```diff
@@ -0,0 +1,142 @@
+                 inputObjs: ArrayBuffer[PrivilegeObject],
+                 commandTypes: Seq[CommandType]): Unit = {
+    commandTypes.foreach {
+      case CommandType.HasChildAsIdentifier =>
+        val query = getFieldVal[LogicalPlan] (plan, "query")
+        buildQuery (query, inputObjs)
+      case _ =>
+    }
+  }
+
+  def buildOutput(plan: LogicalPlan,
+                  outputObjs: ArrayBuffer[PrivilegeObject],
+                  commandTypes: Seq[CommandType],
+                  outputObjsActionType: PrivilegeObjectActionType): Unit = {
+    commandTypes.foreach {
+      case CommandType.HasTableNameAsIdentifier =>
+        val table = invoke(plan, "tableName").asInstanceOf[Identifier]
+        outputObjs += v2TablePrivileges(table)
+
+      case CommandType.HasTableAsIdentifierOption =>
+        val table = getFieldVal[AnyRef](plan, "table")
+        val tableIdent = getFieldVal[Option[Identifier]](table, "identifier")
+        if (tableIdent.isDefined) {
+          outputObjs += v2TablePrivileges(tableIdent.get, actionType = outputObjsActionType)
+        }
+
+      case CommandType.HasTableAsIdentifier =>
+        val table = getFieldVal[LogicalPlan](plan, "table")
+        val tableIdent = getFieldVal[Identifier](table, "identifier")
+        outputObjs += v2TablePrivileges(tableIdent)
+
+      case CommandType.HasChildAsIdentifier =>
+        val table = getFieldVal[AnyRef](plan, "child")
+        val tableIdent = getFieldVal[Identifier](table, "identifier")
+        outputObjs += v2TablePrivileges(tableIdent)
+
+      case _ =>
+    }
+
+  }
+
+  def buildCommandTypes(cmd: String): Seq[CommandType] = {
+    try {
+      val types = invoke(getArcticUtil,
+        "commandType",
+        (StringType.getClass, cmd)).
+        asInstanceOf[Seq[String]]
+      types.map(f => CommandType.withName(f))
+    } catch {
+      case _: Throwable => Seq.empty
+    }
+  }
+
+  def buildOutputActionType(cmd: String): PrivilegeObjectActionType = {
+    try {
+      PrivilegeObjectActionType.withName(invoke(getArcticUtil,
+        "actionType",
+        (StringType.getClass, cmd)).
+        asInstanceOf[String])
+    } catch {
+      case _: Throwable => PrivilegeObjectActionType.OTHER
+    }
+  }
+
+
+  def getArcticUtil: AnyRef = {
+    Class.forName("com.netease.arctic.spark.sql.utils.ArcticAuthUtil")
+  }
+
+  def buildPrivileges(
+                       cmd: String,
+                       plan: LogicalPlan,
+                       inputObjs: ArrayBuffer[PrivilegeObject],
+                       outputObjs: ArrayBuffer[PrivilegeObject]): Unit = {
+    val commandTypes = buildCommandTypes(cmd)
+    val outputActionType = buildOutputActionType(cmd)
+
+    buildInput(plan, inputObjs, commandTypes)
+    buildOutput(plan, outputObjs, commandTypes, outputActionType)
+  }
+
+
+  def accept(commandName: String): Boolean = {
+    try {
+      invoke(
+        Class.forName("com.netease.arctic.spark.sql.utils.ArcticAuthUtil"),
```

Comment:
Recommend to move code in `com.netease.arctic.spark.sql.utils.ArcticAuthUtil` into `ArcticCmdUtils `.
Otherwise, this PR will not have a full picture of Arctic command privilege building and will be hard to review.